### PR TITLE
wasting confusion matrix added

### PR DESCRIPTION
**Pull request by prajurock**
# Description

Weight matrix added similar to height confusion matrix.

User story # (US link)
https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/743/

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests
- [ ] New and existing unit tests pass locally with my changes
- [ ] My Code is review by 1 People

### CODE REVIEW

File: src/common/evaluation/QA/eval_depthmap_models/src/utils.py [hunk 1]
```diff
@@ -371,6 +372,47 @@ def utils(age_in_days, height, sex):
...
+    fig = plt.figure(figsize=(15, 15))
+    ax = fig.add_subplot(111)
+    disp = ConfusionMatrixDisplay(confusion_matrix=data, display_labels=WASTING_DIAGNOSIS)
+    disp.plot(cmap='Blues', values_format='d', ax=ax)
+    s = f"True: {round(T, 2)} False Positive: {round(FP, 2)} False Negative: {round(FN, 2)}"
+    plt.text(0.5, 0.5, s, size=10, bbox=dict(boxstyle="square", facecolor='white'))
+    ax.set_title("Wasting Diagnosis")
+    Path(png_out_fpath).parent.mkdir(parents=True, exist_ok=True)
+    plt.savefig(png_out_fpath)
+    plt.close()
```
Comment (markus-hinsche) [inline src/common/evaluation/QA/eval_depthmap_models/src/utils.py:398]: This code looks quite similar to the stunting code. Can we reduce code duplication?

File: src/common/evaluation/QA/eval_depthmap_models/src/utils.py [hunk 2]
```diff
@@ -371,6 +372,47 @@ def utils(age_in_days, height, sex):
...
+
+
+def calculate_confusion_matrix_wasting(df):
+    cal = Calculator()
+
+    def utils(age_in_days, weight, sex):
```
Comment (markus-hinsche) [inline src/common/evaluation/QA/eval_depthmap_models/src/utils.py:404]: I suggest to change this function's name to be more descriptive, e.g. `calculate_zscore`

**Review by markus-hinsche — COMMENTED** [review]
Once we found a way to speed up the pipeline (https://github.com/Welthungerhilfe/cgm-ml/pull/266) we can address this PR

File: src/common/evaluation/QA/eval_depthmap_models/src/evaluate.py [hunk 3]
```diff
@@ -310,6 +311,14 @@ def get_prediction(model_path: str, dataset_evaluation: tf.data.Dataset) -> np.a
         end = time.time()
         print(f"Total time for Calculate zscores and save confusion matrix: {end - start:.3} sec")
 
+    if WEIGHT_IDX in DATA_CONFIG.TARGET_INDEXES:
+        png_file = f"{OUTPUT_CSV_PATH}/wasting_diagnosis_{RUN_ID}.png"
+        print(f"Calculate and save confusion matrix results to {png_file}")
+        start = time.time()
+        draw_wasting_diagnosis(df, png_file)
+        end = time.time()
+        print(f"Total time for Calculate zscores and save confusion matrix: {end - start:.3} sec")
```
Comment (markus-hinsche) [inline src/common/evaluation/QA/eval_depthmap_models/src/evaluate.py:320]:
```suggestion
        print(f"Total time for Calculate zscores and save wasting confusion matrix: {end - start:.3} sec")
```

File: src/common/evaluation/QA/eval_depthmap_models/src/evaluate.py [hunk 4]
```diff
@@ -310,6 +311,14 @@ def get_prediction(model_path: str, dataset_evaluation: tf.data.Dataset) -> np.a
         end = time.time()
         print(f"Total time for Calculate zscores and save confusion matrix: {end - start:.3} sec")
 
+    if WEIGHT_IDX in DATA_CONFIG.TARGET_INDEXES:
+        png_file = f"{OUTPUT_CSV_PATH}/wasting_diagnosis_{RUN_ID}.png"
+        print(f"Calculate and save confusion matrix results to {png_file}")
```
Comment (markus-hinsche) [inline src/common/evaluation/QA/eval_depthmap_models/src/evaluate.py:316]:
```suggestion
        print(f"Calculate and save wasting confusion matrix results to {png_file}")
```

**Comment by prajurock** [comment]
CODE is WET now

File: src/common/evaluation/QA/eval_depthmap_models/src/utils.py [hunk 5]
```diff
@@ -337,26 +338,16 @@ def draw_stunting_diagnosis(df: pd.DataFrame, png_out_fpath: str):
...
-    plt.text(0.5, 0.5, s, size=10, bbox=dict(boxstyle="square", facecolor='white'))
-    ax.set_title("Stunting Diagnosis")
-    Path(png_out_fpath).parent.mkdir(parents=True, exist_ok=True)
-    plt.savefig(png_out_fpath)
-    plt.close()
+    draw_confusion_matrix(data, png_out_fpath, STUNTING_DIAGNOSIS, 'Stunting Diagnosis')
 
 
-def calculate_confusion_matrix_stunting(df):
+def calculate_zscore_lhfa(df):
```
Comment (markus-hinsche) [inline src/common/evaluation/QA/eval_depthmap_models/src/utils.py:350]: What does "lhfa" mean? Can you spell this out in the function name or add a docstring?

File: src/common/evaluation/QA/eval_depthmap_models/src/utils.py (hunk 2 above)
Comment (markus-hinsche) [reply src/common/evaluation/QA/eval_depthmap_models/src/utils.py:384]: This hasn't been addressed yet

**Review by markus-hinsche — APPROVED** [review]
It would be nice to see the confusion matrix. This will show if the result looks reasonable and the speed is acceptable. 
If both of these things look fine, feel free to merge

**Comment by prajurock** [comment]
image

File: src/common/evaluation/QA/eval_depthmap_models/src/utils.py (hunk 2 above)
Comment (prajurock) [reply src/common/evaluation/QA/eval_depthmap_models/src/utils.py:384]: changed

File: src/common/evaluation/QA/eval_depthmap_models/src/utils.py (hunk 5 above)
Comment (prajurock) [reply src/common/evaluation/QA/eval_depthmap_models/src/utils.py:350]: length height for age

**Pull request merged**